Source/WebCore/rendering/svg/SVGTextLayoutAttributesBuilder.cpp

```diff
@@ -204,31 +196,17 @@ void SVGTextLayoutAttributesBuilder::fillCharacterDataMap(const TextPosition& po
         if (!xListPtr && !yListPtr && !dxListPtr && !dyListPtr && !rotateListPtr)
             break;
 
-        SVGCharacterDataMap::iterator it = m_characterDataMap.find(position.start + i + 1);
-        if (it == m_characterDataMap.end()) {
-            SVGCharacterData data;
-            updateCharacterData(i, lastRotation, data, lengthContext, xListPtr, yListPtr, dxListPtr, dyListPtr, rotateListPtr);
-            m_characterDataMap.set(position.start + i + 1, data);
-            continue;
```

Comment:
This logic has changed. You might want to check if `isNewEntry` when using `HashMap::add` and if so, continue to the next step of the loop.